File: web/scripts/generate-demo-images.py
#!/usr/bin/env python3
"""
Generate "after" demo images with background removed using rembg
Then composite on white background
"""

import os
from pathlib import Path
from rembg import remove
from PIL import Image
import io

# Get the demo directory
script_dir = Path(__file__).parent
demo_dir = script_dir.parent / "public" / "demo"

def process_image(input_file, output_file):
    print(f"Processing {input_file}...")
    
    input_path = demo_dir / input_file
    output_path = demo_dir / output_file
    
    # Read the input image
    with open(input_path, 'rb') as f:
        input_data = f.read()
    
    # Remove background
    print("  Removing background...")
    output_data = remove(input_data)
    
    # Open the result (PNG with transparency)
    img_with_transparency = Image.open(io.BytesIO(output_data))
    
    # Create a white background
    print("  Compositing on white background...")
    white_bg = Image.new('RGBA', img_with_transparency.size, (255, 255, 255, 255))
    
    # Composite the transparent image onto white background
    white_bg.paste(img_with_transparency, (0, 0), img_with_transparency)
    
    # Convert to RGB (no alpha channel needed for final PNG)
    final = white_bg.convert('RGB')
    
    # Save as PNG
    final.save(output_path, 'PNG', quality=95)
    print(f"  Saved to {output_file}")

def main():
    print("Generating demo 'after' images...\n")
    
    for i in range(1, 5):
        try:
            process_image(f"before-{i}.jpg", f"after-{i}.png")
            print()
        except Exception as e:
            print(f"Error processing before-{i}.jpg: {e}")
            print()
    
    print("Done!")

if __name__ == "__main__":
    main()

File: web/scripts/generate-demo-images.mjs
#!/usr/bin/env node
/**
 * Generate "after" demo images with background removed
 * Uses @imgly/background-removal (same as the app)
 */

import { removeBackground } from '@imgly/background-removal';
import sharp from 'sharp';
import { readFile, writeFile } from 'fs/promises';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const demoDir = join(__dirname, '..', 'public', 'demo');

async function processImage(inputFile, outputFile) {
  console.log(`Processing ${inputFile}...`);

  // Read the input image
  const inputPath = join(demoDir, inputFile);
  const imageBuffer = await readFile(inputPath);

  // Create blob from buffer
  const blob = new Blob([imageBuffer], { type: 'image/jpeg' });

  // Remove background using @imgly/background-removal
  console.log('  Removing background...');
  const resultBlob = await removeBackground(blob, {
    progress: (key, current, total) => {
      if (key === 'compute:inference') {
        process.stdout.write(
          `\r  Progress: ${Math.round((current / total) * 100)}%`
        );
      }
    },
  });
  console.log('\n  Background removed');

  // Convert blob to buffer
  const arrayBuffer = await resultBlob.arrayBuffer();
  const pngBuffer = Buffer.from(arrayBuffer);

  // Use sharp to composite onto white background
  console.log('  Compositing on white background...');
  const image = sharp(pngBuffer);
  const metadata = await image.metadata();

  // Create white background and composite the transparent image on top
  const result = await sharp({
    create: {
      width: metadata.width,
      height: metadata.height,
      channels: 4,
      background: { r: 255, g: 255, b: 255, alpha: 1 },
    },
  })
    .composite([{ input: pngBuffer }])
    .png()
    .toBuffer();

  // Save the result
  const outputPath = join(demoDir, outputFile);
  await writeFile(outputPath, result);
  console.log(`  Saved to ${outputFile}`);
}

async function main() {
  console.log('Generating demo "after" images...\n');

  // Process all 4 images
  for (let i = 1; i <= 4; i++) {
    try {
      await processImage(`before-${i}.jpg`, `after-${i}.png`);
      console.log('');
    } catch (error) {
      console.error(`Error processing before-${i}.jpg:`, error.message);
    }
  }

  console.log('Done!');
}

main().catch(console.error);
